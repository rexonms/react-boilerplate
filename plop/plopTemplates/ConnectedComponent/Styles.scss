@import url('https://fonts.googleapis.com/css?family=Dancing+Script:700');

.container {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  height: 100vh;
}
.content {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 600px;
  height: 400px;
  margin: auto;
  padding: 40px;
  background: #111;
  border-radius: 20px;
  p {
  }
}
.heading {
  font-family: 'Dancing Script', cursive;
  font-size: 80px;
  margin: 20px 0 40 0;
}