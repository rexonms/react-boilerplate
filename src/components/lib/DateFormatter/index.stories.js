import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import DateFormatter from './index';

const stories = storiesOf('components/lib/DateFormatter', module);
const props = {
  timestamp: 1527633993,
  country: 'US',
};
const countries = ['US', 'GB', 'CA'];
stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('DateFormatter', withInfo(`
  Formats epoch timestamp to date based on country code
  ~~~js
    <DateFormatter timestamp={1527633993} country="US" />
  ~~~
`)(() => {
  const countryOptions = select('country', countries, countries[0]);
  return <DateFormatter timestamp={props.timestamp} country={countryOptions} />;
}));
