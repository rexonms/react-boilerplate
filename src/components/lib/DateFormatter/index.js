import React from 'react';
import PropTypes from 'prop-types';

import Styles from './DateFormatter.scss';

const propTypes = {
  timestamp: PropTypes.number.isRequired,
  country: PropTypes.string,
  yearFormat: PropTypes.string,
};

const defaultProps = {
  country: 'US',
  yearFormat: 'YY',
};

const setDoubleDigits = (num) => {
  let newNum = num.toString();
  newNum = newNum.length === 1 ? `0${newNum}` : newNum;
  return newNum;
};

// https://en.wikipedia.org/wiki/Date_format_by_country
export const ymdCountries = ['AF', 'AU', 'BT', 'CA', 'CN', 'CU', 'HK', 'HU', 'IN', 'IR', 'JM', 'JP', 'KP', 'KR', 'MO', 'MV', 'MN', 'MM', 'NP', 'NO', 'SG', 'ZA', 'LK', 'SE', 'TW', 'UZ'];
export const mdyCountries = ['US', 'AS', 'BS', 'FM', 'Gl', 'GU', 'KE', 'MH', 'MP', 'PK', 'PA', 'PH', 'PR', 'SO', 'TG', 'UM'];
const getDate = (props) => {
  const d = new Date(props.timestamp * 1000);
  let yy = d.getUTCFullYear();
  if (props.yearFormat === 'YY') {
    yy = yy.toString().substring(2, 4);
  }
  const mm = setDoubleDigits(d.getUTCMonth() + 1);
  const dd = setDoubleDigits(d.getUTCDate());
  if (mdyCountries.includes(props.country)) {
    return `${mm}/${dd}/${yy}`;
  } else if (ymdCountries.includes(props.country)) {
    return `${yy}/${mm}/${dd}`;
  }
  return `${dd}/${mm}/${yy}`;
};

const DateFormatter = props => (
  <div className={`${Styles.container}`}>
    <p>{getDate(props)}</p>
  </div>
);

DateFormatter.propTypes = propTypes;
DateFormatter.defaultProps = defaultProps;

export default DateFormatter;
