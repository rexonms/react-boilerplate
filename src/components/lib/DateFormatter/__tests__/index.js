import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import DateFormatter from '../index';
import Styles from '../DateFormatter.scss';

describe('DateFormatter Component', () => {
  let wrapper;
  const props = {
    timestamp: 1527633993,
  };
  const YMD = '18/05/29';
  const DMY = '29/05/18';
  const MDY = '05/29/18';

  beforeEach(() => {
    wrapper = shallow(<DateFormatter {...props} />);
  });

  it('renders matchs previous snapshot', () => {
    const tree = renderer.create(<DateFormatter timestamp={1527633993} country="US"  />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with proper className', () => {
    expect(wrapper.exists(Styles.container)).toBe(true);
  });

  it('renders with only time stamp', () => {
    expect(wrapper.text()).toBe(MDY);
  });

  it('formats CA date correctly ', () => {
    props.country = 'CA';
    wrapper = shallow(<DateFormatter {...props} />);
    expect(wrapper.text()).toBe(YMD);
  });

  it('formats CN date correctly ', () => {
    props.country = 'CN';
    wrapper = shallow(<DateFormatter {...props} />);
    expect(wrapper.text()).toBe(YMD);
  });

  it('formats DE date correctly ', () => {
    props.country = 'DE';
    wrapper = shallow(<DateFormatter {...props} />);
    expect(wrapper.text()).toBe(DMY);
  });

  it('formats AS date correctly ', () => {
    props.country = 'AS';
    wrapper = shallow(<DateFormatter {...props} />);
    expect(wrapper.text()).toBe(MDY);
  });

  it('formats IT date correctly ', () => {
    props.country = 'IT';
    wrapper = shallow(<DateFormatter {...props} />);
    expect(wrapper.text()).toBe(DMY);
  });

  it('renders with properly for YYYY format', () => {
    const output = '05/29/2018';
    props.yearFormat = 'YYYY';
    props.country = 'US';
    wrapper = shallow(<DateFormatter {...props} />);
    expect(wrapper.text()).toBe(output);
  });

});
