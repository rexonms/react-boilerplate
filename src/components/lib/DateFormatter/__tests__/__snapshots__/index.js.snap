// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateFormatter Component renders matchs previous snapshot 1`] = `
<div
  className="container"
>
  <p>
    05/29/18
  </p>
</div>
`;
